remo/events/forms.py

```diff
@@ -185,20 +182,23 @@ def __init__(self, *args, **kwargs):
         self.user = kwargs.pop('user', None)
         super(EventForm, self).__init__(*args, **kwargs)
 
-        # Dynamic categories/goals field.
+        # Dynamic categories field.
+        categories_query = FunctionalArea.objects.filter(Q(active=True))
+
         if self.instance.id:
-            categories_query = (Q(active=True) |
-                                Q(id__in=self.instance.categories.all()))
-            goals_query = Q(active=True) | Q(id__in=self.instance.goals.all())
-            categories = FunctionalArea.objects.filter(categories_query)
-            goals = EventGoal.objects.filter(goals_query)
-            self.fields['categories'].queryset = categories
-            self.fields['goals'].queryset = goals
+            categories_query |= categories_query.filter(Q(
+                id__in=self.instance.categories.all()))
+            initial_category = self.instance.categories.all()[0]
```

Comment:
You might need to try/catch this or check if queryset exists (I am getting errors on some old events).